### ISSUE TITLE
convert to `PosixPath`

### ISSUE THREAD

**Issue by hasufell** [haskell/tar#78]
caveats:

* on windows we'll have to assume UTF-8
* we probably should wait for https://github.com/haskell/filepath/pull/202 (or `os-string` package) to get cleaner access to certain functions

context: https://github.com/haskell/tar/issues/6#issuecomment-1817491759

**Comment by Bodigrim** [comment]
I did as much work as possible without changing public API (and thus without a major version bump) in dfc8fc4 and c2c70ce. This would suffice to support Unicode filenames though.

**Comment by hasufell** [comment]
Yeah, but it doesn't support non-unicode on unix. I did not have time to look at anything. Feel free to rebase this on master.
